code/modules/mob/living/carbon/human/damage_menu.dm

```diff
@@ -0,0 +1,151 @@
+// the damage menu is a menu which you can use to damage human mobs
+// it's accessible via the drop down list when you VV a human mob
+
+/datum/vueui_module/damage_menu
+	var/datum/weakref/target_human
+
+/datum/vueui_module/damage_menu/New(var/datum/weakref/target, var/mob/user)
+	var/mob/living/carbon/human/H = target.resolve()
+	if(istype(H))
+		target_human = target
+	ui_interact(user)
+
+/datum/vueui_module/damage_menu/ui_interact(mob/user)
+	if(!user.client.holder)
+		return
+	var/datum/vueui/ui = SSvueui.get_open_ui(user, src)
+	if(!ui)
+		var/human_name = "Geoff"
+		var/human_ckey = "None"
+		var/mob/living/carbon/H = target_human.resolve()
+		if(H)
+			human_name = H.real_name
+			if(H.ckey)
+				human_ckey = H.ckey
+		ui = new(user, src, "admin-damage-menu", 600, 600, "Damage Menu | [human_name] | [human_ckey]", state = interactive_state)
+		ui.header = "minimal"
+	ui.open()
+
+/datum/vueui_module/damage_menu/vueui_data_change(var/list/data, var/mob/user, var/datum/vueui/ui)
+	if(!data)
+		data = list()
+	if(!user.client.holder)
+		return
+
+	var/mob/living/carbon/H = target_human.resolve()
+	if(!istype(H))
+		to_chat(user, SPAN_DANGER("The humanoid target couldn't be found, closing UI."))
+		ui.close()
+
+	var/list/limbs = list()
+	for(var/name in H.species.has_limbs)
+		limbs[capitalize_first_letters(parse_zone(name))] = !!H.organs_by_name[name]
+	var/list/organs = list()
+	for(var/name in H.species.has_organ)
+		organs[capitalize_first_letters(name)] = !!H.internal_organs_by_name[name]
+
+	data["limbs"] = limbs
+	data["organs"] = organs
+
+	return data
+
+/datum/vueui_module/damage_menu/Topic(href, href_list)
+	if(!usr.client.holder)
+		return
+
+	var/mob/living/carbon/human/H = target_human.resolve()
+	if(!istype(H))
+		return
+
+	switch(href_list["target"])
+		if("limb")
+			var/obj/item/organ/external/L = H.organs_by_name[reverse_parse_zone(lowertext(href_list["name"]))]
```

Comment:
I have to say, this is the ugliest shit I've ever seen

Reply:
self insert